### ISSUE TITLE
No timezone in the syslog

### ISSUE THREAD

**Issue by ADeeeee** [CommunityHoneyNetwork/hpfeeds-logger#30]
OS: Ubuntu 18.04.4 LTS
Docker: Docker version 19.03.6, build 369ce74a3c
Python: 2.7.17

I am not sure where the root cause is (Graylog or no timezone in syslog) so far, so I create issues for both, but it's still good to follow the rule in reference below.

**Issue:**
> 2020-02-14T09:15:10.855896
 
This timestamp is an output from the formatters which has no timezone at all. It's in UTC for sure, but if you output the syslog remotely to Graylog which is data in UTC and non-UTC in graylog's host combined, the timestamp will be transformed to UTC automatically. 

Which causes an issue in my case here for example:

> syslog: 2020-02-14T09:15:10
> Graylog Host: 2020-02-14T17:15:10   << my local timezone, UTC+8
> Raw datain Graylog: 2020-02-14T01:15:10  << got transformed by default in Graylog

Could we have the timezone in the output like:
> 2020-02-14T09:15:10.855896+00:00
or
> 2020-02-14T09:15:10.855896Z

or even has an argument in hpfeeds.logger.sysconfig which can make the magic in different timezones in the future?

**Ref:**
https://stackoverflow.com/questions/19654578/python-utc-datetime-objects-iso-format-doesnt-include-z-zulu-or-zero-offset/54272238

**Workaround:**
I'm planning to build the image myself and change the code in https://github.com/CommunityHoneyNetwork/hpfeeds-logger/tree/master/hpfeeds-logger/hpfeedslogger/formatters
from
`timestamp = datetime.datetime.isoformat(datetime.datetime.utcnow())`
to
`timestamp = datetime.datetime.isoformat(datetime.datetime.now())`

so that I can set the timezone in container and get it right.

**Comment by JesseBowling** [comment]
Hi! Thanks for reporting; we’ll take a look at this. 

Right now we’re rolling in changes to containers to ensure they’re pure Python 3; please test any source built images thoroughly as we just did early merges. 🙂

**Comment by ADeeeee** [comment]
Hi Jesse,

I did built an image and it worked for me. This helps me to change the timezone in the container of hpfeeds-logger and get the right time in my timezone from python.

Here is the diff in hpfeeds-logger besides the files of formatters:

Dockerfile
```
+   ENV TZ=YOUR_TIMEZONE
-    && apt-get install -y ansible
+    && apt-get install -y ansible tzdata && dpkg-reconfigure -f noninteractive tzdata
```

update:
There's temporary name resolution problem in the master branch, so I checkout back to v1.8 and it works.

I guess it'd be no problem if there's timezone in the syslog already, but no resource to try it yet. Just FYI in python3 and hope it will be in the future patch: https://stackoverflow.com/questions/3401428/how-to-get-an-isoformat-datetime-string-including-the-default-timezone

**Comment by JesseBowling** [comment]
Hi @ADeeeee ! Please check out the changes merged from #31 . This may or may not help your specific goals.

**Comment by ADeeeee** [comment]
Hi Jesse,

I might have some problems with the error below because I'm running other containers with v1.8 in production, so I just modified the hpfeeds-logger/hpfeedslogger/formatters/splunk.py on v1.8 branch and built another image, then use it in the same environment.

> Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/dist-packages/hpfeeds/client.py", line 88, in tryconnect
>     return self.connect()
>   File "/usr/local/lib/python3.6/dist-packages/hpfeeds/client.py", line 114, in connect
>     socket.SOCK_STREAM,
>   File "/usr/lib/python3.6/socket.py", line 745, in getaddrinfo
>     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
> socket.gaierror: [Errno -3] Temporary failure in name resolution


**Result:**
Which works great! UTC in container and in syslog with 'Z' correctly. Graylog did NOT transform it again as we expected. 

Just really not used to it when the timestamp is not in front of the line anymore while reading "logs", which I'd prefer putting the timestamp in front of the lines in syslog mode like before.

Thanks!
